part-2.py

```diff
@@ -5,12 +5,43 @@
 
 # search
 
-
+def search(array, query):
+    for item in array:
+        if isinstance(item, list):
+            return search(array, query)
+        if query in array:
+            return True
+    else:
+        return False
```

Comment:
The else for the `for` isn't needed.

```suggestion
     return False
```